app/validators/slug_validator.rb

```diff
@@ -129,6 +130,18 @@ def validate!
     end
   end
 
+  class DetailedGuideValidator <InstanceValidator
+
+    def applicable?
+      of_kind?('detailed_guide')
+    end
+
+    def validate!
+      record.errors[attribute] << "must be a valid slug under guidance/" unless value.match(%r{^(guidance/)?[a-z0-9\-_]+$})
```

Comment:
I understand given the TODO in the test that this is going to change, but your error message does not match your validation regex (in that the `guidance` prefix is currently optional).

Also could you change the whitespaceing in this class to match the other validators?  i.e. fewer newlines, `< InstanceValidator` instead of `<InstanceValidator`.